metrics.go

```diff
@@ -4,43 +4,36 @@ import (
 	"context"
 	"time"
 
+	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/attribute"
-	"go.opentelemetry.io/otel/metric/global"
-	"go.opentelemetry.io/otel/metric/instrument"
-	"go.opentelemetry.io/otel/metric/unit"
+	"go.opentelemetry.io/otel/metric"
 )
 
-var meter = global.MeterProvider().Meter("header")
+var meter = otel.Meter("header")
 
 // WithMetrics enables Otel metrics to monitor head and total amount of synced headers.
 func WithMetrics[H Header](store Store[H]) error {
-	headC, _ := meter.AsyncInt64().Counter(
+	headC, _ := meter.Int64ObservableCounter(
 		"head",
-		instrument.WithUnit(unit.Dimensionless),
-		instrument.WithDescription("Subjective head of the node"),
+		metric.WithDescription("Subjective head of the node"),
 	)
 
-	err := meter.RegisterCallback(
-		[]instrument.Asynchronous{
-			headC,
-		},
-		func(ctx context.Context) {
-			// add timeout to limit the time it takes to get the head
-			// in case there is a deadlock
-			ctx, cancel := context.WithTimeout(ctx, time.Second)
-			defer cancel()
+	callback := func(ctx context.Context, observer metric.Observer) error {
+		// add timeout to limit the time it takes to get the head
+		// in case there is a deadlock
+		ctx, cancel := context.WithTimeout(ctx, time.Second)
+		defer cancel()
 
-			head, err := store.Head(ctx)
-			if err != nil {
-				headC.Observe(ctx, 0, attribute.String("err", err.Error()))
-				return
-			}
-
-			headC.Observe(
-				ctx,
-				head.Height(),
-			)
-		},
-	)
+		head, err := store.Head(ctx)
+		if err != nil {
+			observer.ObserveInt64(headC, 0,
+				metric.WithAttributes(
+					attribute.String("err", err.Error())))
+		} else {
+			observer.ObserveInt64(headC, head.Height())
+		}
+		return nil
```

Comment:
shouldn't we return err?

Reply:
I guess because it's not the error coming from metrics